log/log.go

```diff
@@ -0,0 +1,114 @@
+package log
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+)
+
+type loggingLevel int
+
+// Log messages can have one of 3 possible states.
+const (
+	DEBUG loggingLevel = 1 << iota
+	INFO
+	CRITICAL
+)
+
+// List of fields that should not be overwritten by fields specified when logging
+// a message
+var restrictedFields = []string{"level", "time", "message"}
+
+// Creates a new logging instance
+func New(out io.Writer, level loggingLevel, fields map[string]interface{}) *Logger {
+	return &Logger{
+		out:           out,
+		defaultFields: fields,
+		level:         level,
+	}
+}
+
+type Logger struct {
+	// Create a lock when writing to `out` so output is not intermingled
+	mu  sync.Mutex
+	out io.Writer
+	// List of default fields that will be serialized
+	// along with the message and written to `out`
+	defaultFields map[string]interface{}
+	level         loggingLevel
+}
+
+// Log a debug message as long as the logger's debug level is set to at least Debug
+func (l *Logger) Debug(message string, fields map[string]interface{}) {
```

Comment:
Proposed interface to hide logrus behind:

https://public.etherpad-mozilla.org/p/logrus-simplified